### PR TITLE
Create Broker Form Component

### PR DESCRIPTION
**Pull request by Msarawan**
Broker Form Component :

Screenshot from 2023-05-30 00-10-37

### CODE REVIEW

**Comment by ajaypratap003** [comment]
@Msarawan  Form name filed should not be take full width of page. It should take half width of page. You can use layout like Grid etc.  Buttons are missing in form i.e. `Create` and `Cancel`
On cancel, user should redirect on brokers list page , it should be done by routing. 

See below screenshot for reference.
image

File: src/brokers/add-broker/components/AddBrokerForm/FormPage.component.tsx [hunk 1]
```diff
@@ -1,38 +1,81 @@
...
+  Grid,
+  GridItem,
+  ActionListItem,
+  ActionList,
+} from '@patternfly/react-core';
 import { useTranslation } from '../../../../i18n';
+import { Link } from 'react-router-dom';
+type validate = 'success' | 'warning' | 'error' | 'default';
 
 export const FormPage: FC = () => {
```
Comment (ajaypratap003) [inline src/brokers/add-broker/components/AddBrokerForm/FormPage.component.tsx:18]: File and component name  `FormView` would be better.

Comment (Msarawan) [reply src/brokers/add-broker/components/AddBrokerForm/FormPage.component.tsx:18]: okay Ajay i will change the name.

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json [hunk 2]
```diff
@@ -48,5 +48,6 @@
   "broker": "Broker",
   "configure_via": "Configure via:",
   "form_view": "Form view",
-  "yaml_view": "YAML view"
+  "yaml_view": "YAML view",
+  "info_alert": "Note: Some fields may not be represented in this form view. Please select 'YAML view' for full control."
```
Comment (ajaypratap003) [inline locales/en/plugin__activemq-artemis-self-provisioning-plugin.json:52]:
```suggestion
  "create_broker_form_info_alert": "Note: Some fields may not be represented in this form view. Please select 'YAML view' for full control."
```

File: src/brokers/add-broker/AddBroker.container.tsx [hunk 3]
```diff
@@ -20,7 +21,7 @@ const AddBrokerPage: FC<AddBrokerPageProps> = ({ match }) => {
     apiVersion: 'broker.amq.io/v1beta1',
     kind: 'ActiveMQArtemis',
     metadata: {
-      name: 'default',
+      name: '',
```
Comment (ajaypratap003) [inline src/brokers/add-broker/AddBroker.container.tsx:24]:
```suggestion
      name: 'default',
```

Let's keep the default value of  name field.

File: src/brokers/add-broker/AddBroker.container.tsx [hunk 4]
```diff
@@ -32,9 +33,21 @@ const AddBrokerPage: FC<AddBrokerPageProps> = ({ match }) => {
     },
   };
 
+  const handleNameChange = (fieldName: string, value: string) => {
```
Comment (ajaypratap003) [inline src/brokers/add-broker/AddBroker.container.tsx:36]:
```suggestion
  const onChangeField = (fieldName: string, value: string) => {
```

File: src/brokers/add-broker/components/AddBrokerForm/FormView.component.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    e.preventDefault();
+    if (data.metadata.name) {
+      setShowAlert(true);
+    }
+  };
+
+  return (
+    <Grid className="pf-u-mx-md">
+      <GridItem span={6}>
+        {showAlert && <Alert variant="info" isInline title={t('info_alert')} />}
```
Comment (ajaypratap003) [inline src/brokers/add-broker/components/AddBrokerForm/FormView.component.tsx:38]:
```suggestion
       <Alert variant="info" isInline title={t('info_alert')} />
```
This alert should display by default.We don't need to add any check.

Comment (Msarawan) [reply src/brokers/add-broker/components/AddBrokerForm/FormView.component.tsx:38]: okay

**Comment by ajaypratap003** [comment]
closing this PR against new PR https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/pull/47

**Pull request closed without merging**